custom_components/waste_collection_schedule/waste_collection_schedule/source/midsussex_gov_uk.py

```diff
@@ -1,154 +1,162 @@
+        # We clean and store the inputs
+        self._number = number.strip()
+        self._street = street.strip()
+        self._postcode = postcode.strip().replace(" ", "+") # URL-encode space
+
+    def fetch(self) -> list[Collection]:
+        entries = []
+        BASE_URL = "https://sms-wrp.whitespacews.com"
+
+        with requests.Session() as session:
+            # --- STEP 1: Get the Dynamic Track Token ---
+            r1 = session.get(f"{BASE_URL}/mop.php")
+            r1.raise_for_status()
+
+            soup = BeautifulSoup(r1.text, 'html.parser')
+
+            next_link_element = soup.find('a', href=lambda href: href and 'Track=' in href)
+
+            if not next_link_element:
+                 raise Exception("Could not find dynamic Track token link in Step 1.")
+
+            dynamic_link = next_link_element.get('href')
+
             try:
-                normal_date = datetime.strptime(tds[0].text.strip(), "%A %d %B").date()
-                festive_date = datetime.strptime(tds[1].text.strip(), "%A %d %B").date()
-            except Exception:
-                continue
-            for entry in entries.copy():
-                date = entry.date
-                if date.month == normal_date.month and date.day == normal_date.day:
-                    entries.remove(entry)
-                    entries.append(
-                        Collection(
-                            date=festive_date.replace(year=date.year),
-                            t=entry.type,
-                            icon=entry.icon,
-                        )
+                track_token = dynamic_link.split('Track=')[1].split('&')[0]
+            except IndexError:
+                raise Exception("Could not parse dynamic Track token in Step 1.")
+
+            # --- STEP 2: Submit the Address ---
+            post_url = f"{BASE_URL}/mop.php?serviceID=A&Track={track_token}&seq=2"
+
+            payload = {
+                'address_name_number': self._number,
+                'address_street': self._street,
+                'street_town': '',
+                'address_postcode': self._postcode.replace('+', ' '), # Send post data un-encoded
+            }
+
+            r2 = session.post(post_url, data=payload)
+            r2.raise_for_status()
+
+            # --- STEP 3: Select the Specific Address (Get pIndex) ---
+            soup2 = BeautifulSoup(r2.text, 'html.parser')
+
+            search_text = self._number.upper()
+
+            address_link = soup2.find(
+                'a',
+                class_='app-subnav__link',
+                string=lambda t: t and search_text in t.upper()
+            )
+
+            if not address_link:
+                if soup2.find(string=lambda t: t and "address not listed" in t.lower()):
+                    raise SourceArgumentException(
+                        "Address not found. Check house name/number and postcode.",
+                        argument="number"
                     )
```

Comment:
Please look at the classes before you use them

```py
class SourceArgumentException(Exception):
  def __init__(self, argument: str, message: str) -> None:
```

so this code generates a `SourceArgumentException.__init__() got multiple values for argument 'argument'`

I'd suggest raising a `SourceArgumentNotFound` here


```suggestion
                    raise SourceArgumentNotFound(
                        "number",
                        self._number
                    )
```